src/main.ts

```diff
@@ -35,25 +35,71 @@ export const main = async (indices?: number[]) => {
     console.log('main function has completed');
 };
 
+function indexGenerator(): number | null {
+    // This function is not pure, its return value depends on the time of the day.
+    // It returns ether a number or null.
+    // null signifies that a file should not be generated at this time.
+    // Otherwise the number id a day index of the file.
+
+    // Rules:
+    // - Rule 1: We generate files from index 2 to index 14 (meaning [today]+2 to [today]+14)
+    // - Rule 2: We start generation after 1am to let holiday stops finish.
+    //           In other words, we do not generate during hour 0 of the day
+    // - Rule 3: We only generate index 2 before 10am.
+
+    // The mapping is:
+    // Hour 00 -> dayIndex = 1 (index 1 is not generated) Rule 1 and Rule 2
+    // Hour 01 -> dayIndex = 2
+    // Hour 02 -> dayIndex = 3
+    // Hour 03 -> dayIndex = 4
+    // Hour 04 -> dayIndex = 5
+    // Hour 05 -> dayIndex = 6
+    // Hour 06 -> dayIndex = 7
+    // Hour 07 -> dayIndex = 8
+    // Hour 08 -> dayIndex = 9
+    // Hour 09 -> dayIndex = 10
+    // Hour 10 -> dayIndex = 11
+    // Hour 11 -> dayIndex = 12
+    // Hour 12 -> dayIndex = 13
+    // Hour 13 -> dayIndex = 14
+    // Hour 14 -> dayIndex = 1 (index 1 is not generated) Rule 1
+    // Hour 15 -> dayIndex = 2 (index 2 is not generated after 10am) Rule 3
+    // Hour 16 -> dayIndex = 3
+    // Hour 17 -> dayIndex = 4
+    // Hour 18 -> dayIndex = 5
+    // etc...
+
+    const dayIndex = 1 + (new Date().getUTCHours() % 14);
```

Comment:
Should this be hours?

Reply:
We derive a day index from the current hour. Yes.

Reply:
So it generates one day fulfilment file per hour? One file at a time

Reply:
Yep. The lambda is hourly. And that is reflected in the file generation time stamps in S3 :) We could not generate all files at once, because Zuora doesn't support it well.

Reply:
>because Zuora doesn't support it well.

that was a rumour that got out of hand, I'm not sure anyone actually checked and proved that the queries needed to be spaced an hour apart.

Reply:
😄

Reply:
For the record, a previous version of the lambda was generating all 14 files in one execution and that was working well :)